### PR TITLE
Allow unsetting profile images and description in updateProfile

### PR DESCRIPTION
**Pull request by devinivy**
A few steps here, in order to allow unsetting profile images and descriptions in the `app.bsky.actor.updateProfile` method.
 - Add support for `nullable` on lexicon object types, spec'ed similarly to `required`.
 - Add `nullable` to codegen.
 - Alter lexicons and implementation of `updateProfile` to allow unsetting avatar, banner, and description.

### CODE REVIEW

**Comment by pfrazee** [comment]
Nice. I think this is the right call and the PR lgtm

File: packages/lex-cli/src/codegen/lex-gen.ts [hunk 1]
```diff
@@ -127,11 +129,17 @@ export function genObject(
...
-              )}[]`,
+              type: makeType(
+                refToType(
+                  propDef.items.ref,
+                  stripScheme(stripHash(lexUri)),
+                  imports,
+                ),
+                {
+                  nullable: propNullable,
+                  array: true,
```
Comment (dholms) [inline packages/lex-cli/src/codegen/lex-gen.ts:140]: just out of curiosity, what does this `array: true` do?

Comment (devinivy) [reply packages/lex-cli/src/codegen/lex-gen.ts:140]: An example of that would be the difference between `A | B` and `(A | B)[]`.

**Review by dholms — APPROVED** [review]
awesome 👍 

feels very natural in lex

**Pull request merged**